java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/jobexecution/CompactSortedFiles.java

```diff
@@ -286,14 +286,13 @@ private static void updateStateStoreSuccess(List<String> inputFiles,
             FileInfo fileInfo = FileInfo.wholeFile()
                     .filename(file)
                     .partitionId(partitionId)
-                    .fileStatus(FileInfo.FileStatus.READY_FOR_GARBAGE_COLLECTION)
+                    .numberOfRecords(recordsWritten)
```

Comment:
This is incorrect, but it seems like a problem with the method signature for atomicallyUpdateFilesToReadyForGCAndCreateNewActiveFile.

It could take a partition ID and list of filenames to mark ready for GC, rather than a list of FileInfo. Maybe that should be a separate issue.

Reply:
Thanks, created https://github.com/gchq/sleeper/issues/1633